pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+    n_x_grid = _find_next_grid_1d(point[0], vec[0], bounds[0])
+    n_y_grid = _find_next_grid_1d(point[1], vec[1], bounds[1])
+    n_z_grid = _find_next_grid_1d(point[2], vec[2], bounds[2])
+    next_grid = np.array([n_x_grid, n_y_grid, n_z_grid])
+    # Divide by direction cosines to calculate the distance along vec
+    dist_maxs = np.divide(next_grid - point, vec)
+    return dist_maxs
+
+def _move_to_next_voxel(idxs, point, dist_temp, dist_maxs, vec):
+    """
+    Move the point to next voxel boundary.
+
+    Parameters:
+    -----------
+    idxs: list
+        List of indexes, format: [x_idx, y_idx, z_idx]
+    point : numpy array
+        An array of size 3. Current point position.
+    dist_temp: float
+        Temporary travel length from the start.
+    dist_maxs: numpy array
+        The maximum travel length in current voxel for 3 directions.
+    vec : numpy array
+        Direction vector from start to end.
+
+    Returns:
+    --------
+    idxs: list
+        Updated indexes.
+    updated_point : numpy array
+        An array of size 3. Updated point positon.
+    dist_temp : float
+        Updated travel length since the beginning of the ray."""
+
+    dist_min = min(dist_maxs[dist_maxs>0])
+    update_dir = list(dist_maxs).index(dist_min)
+    # vec[update_dir] will not be 0
+    if vec[update_dir] > 0:
+        idxs[update_dir] += 1
+    elif vec[update_dir] < 0:
+        idxs[update_dir] -= 1 
+    updated_point = np.add(point, dist_min * vec)
+    dist_temp += dist_min
+    return idxs, updated_point, dist_temp
+
+def _is_ray_on_boundary(start, vec, bounds):
+    """
+    Check whether the ray on a boundary surface.
+
+    Parameters:
+    -----------
+    start : numpy array
+        Start point coordinate.
+    vec : numpy array
+        Direction vector.
+    bounds : numpy array
+        Boundaries of the mesh grid.
+
+    Return:
+    -------
+    True : bool
+        Ray is on a boundary surface
+    False : bool
+        Ray is not on a boundary surface
+    """
+    if 1.0 in vec:
+        for idx in np.argwhere(vec!=1.0).flatten():
+            if start[idx] in bounds[idx]:
+                return True
+    return False
+
+def _ray_voxel_traverse(bounds, start, end):
+    """
+    Ray (line segment) traversal in structured mesh.
+    Return the voxels that intersect with the line segment.
+
+    Parameters:
+    -----------
+    bounds : numpy array
+        Boundaries of the mesh grid.
+    start : numpy array
+        Start point.
+    end : numpy array
+        End point.
+
+    Return:
+    -------
+    voxles : list of tuples
+        List of voxel indices. These voxels intersect with the line segment.
+    """
+    # Initialization
+    # Find the voxel that start point in
+    point = np.copy(start)
+    idxs = [-1, -1, -1]
+    voxels = set()
+    vec = _calc_vec_dir(start, end)
+    # if the ray on a boundary, then return empty set
+    if _is_ray_on_boundary(start, vec, bounds):
+        return set()
```

Comment:
for consistency, why not just return "voxels" since this is the first check after creating the set it'll be empty as you needed it to be returned if the if statement is True.
```suggestion
        return voxels
```